python/pyspark/sql/dataframe.py

```diff
@@ -78,6 +78,12 @@ def __init__(self, jdf, sql_ctx):
         self.is_cached = False
         self._schema = None  # initialized lazily
         self._lazy_rdd = None
+        self._eager_eval = sql_ctx.getConf(
+            "spark.jupyter.eagerEval.enabled", "false").lower() == "true"
+        self._default_console_row = int(sql_ctx.getConf(
+            "spark.jupyter.default.showRows", u"20"))
+        self._default_console_truncate = int(sql_ctx.getConf(
+            "spark.jupyter.default.showRows", u"20"))
```

Comment:
I guess we shouldn't hold these three values but extract as `@property` or refer each time in `_repr_html_`. Otherwise, we'll hit unexpected behavior, e.g.:

```python
df = ...
spark.conf.set("spark.jupyter.eagerEval.enabled", True)
df
```

won't show the html.

Reply:
Yep, I'll fix it in next commit. 
![image](https://user-images.githubusercontent.com/4833765/40306628-fecbc1aa-5d31-11e8-970d-cc1bcd0e8952.png)